lnbits/core/crud.py

```diff
@@ -619,6 +627,63 @@ async def update_payment_extra(
     )
 
 
+async def update_pending_payments(wallet_id: str):
+    pending_payments = await get_payments(
+        wallet_id=wallet_id,
+        pending=True,
+        exclude_uncheckable=True,
+    )
+    for payment in pending_payments:
+        await payment.check_status()
+
+
+async def get_payments_history(
+    wallet_id: Optional[str] = None,
+    group: DateTrunc = "day",
+    filters: Optional[Filters] = None,
+) -> List[PaymentHistoryPoint]:
+    if not filters:
+        filters = Filters()
+    where = ["(pending = False OR amount < 0)"]
+    values = []
+    if wallet_id:
+        where.append("wallet = ?")
+        values.append(wallet_id)
+
+    transactions = await db.fetchall(
+        f"""
+        SELECT {db.truncate_date('time', group)} date,
```

Comment:
An equivalent of `db.truncate_date` should be applied in the frontend (chart) that needs the truncated time stamps, not in the db level.